File: serene/src/Serene.Web/Modules/Administration/User/UserColumns.cs
namespace Serene.Administration.Columns;

[ColumnsScript("Administration.User")]
[BasedOnRow(typeof(UserRow), CheckNames = true)]
public class UserColumns
{
    [EditLink, AlignRight, Width(55)]
    public string UserId { get; set; }
    [EditLink, Width(150)]
    public string Username { get; set; }
    [Width(150)]
    public string DisplayName { get; set; }
    [Width(250)]
    public string Email { get; set; }
    [Width(80)]
    public string Source { get; set; }
    [QuickFilter, Width(300)]
    public string Roles { get; set; }
}

File: serene/src/Serene.Web/Modules/ServerTypes/Administration/UserColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UserRow } from "./UserRow";

export interface UserColumns {
    UserId: Column<UserRow>;
    Username: Column<UserRow>;
    DisplayName: Column<UserRow>;
    Email: Column<UserRow>;
    Source: Column<UserRow>;
    Roles: Column<UserRow>;
}

export class UserColumns extends ColumnsBase<UserRow> {
    static readonly columnsKey = 'Administration.User';
    static readonly Fields = fieldsProxy<UserColumns>();
}